realm/src/main/java/io/realm/Realm.java

```diff
@@ -689,11 +689,13 @@ private static void validateAgainstExistingConfigurations(RealmConfiguration new
     private static void initializeRealm(Realm realm) {
         long version = realm.getVersion();
         boolean commitNeeded = false;
```

Comment:
I thought so. But there is a chance setVersion throw an exception and the transaction needs to be cancelled? Would this be a case?